components/notion/actions/common/base-page-builder.mjs

```diff
@@ -161,5 +161,63 @@ export default {
     createBlocks(pageContent) {
       return markdownToBlocks(pageContent);
     },
+    /**
+     * Formats the children of an existing block for creating/appending
+     * to a new page/block
+     */
+    async formatChildBlocks(block) {
+      const children = block.children;
+      if (!block.has_children) {
+        return [];
+      }
+
+      (await Promise.all(children.map((child) => this.formatChildBlocks(child))))
+        .forEach((c, i) => {
+          const child = children[i];
+          children[i] = {
+            object: "block",
+            type: child.type,
+            [child.type]: child[child.type],
+          };
+          if (c.length > 0) {
+            children[i][child.type].children = c;
```

Comment:
Observation: I'm not against this kind of code but at the same time not a big fan of it because it's changing references inside `children` object array which might be a source of issues in the future.